packages/react/src/components/password-creation-input/password-creation-input.tsx

```diff
@@ -0,0 +1,164 @@
+import { v4 as uuid } from '../../utils/uuid';
+
+const StyledFieldContainer = styled(FieldContainer)`
+    > :nth-child(2) {
+        margin-bottom: 0;
+    }
+`;
+
+const StyledUl = styled.ul`
+    font-size: 0.75rem;
+    margin: 0 0 var(--spacing-half) 0;
+    padding: 0;
+`;
+
+const PasswordContainer = styled.div`
+    margin-bottom: calc(var(--spacing-1x) * 1.5);
+    position: relative;
+`;
+
+export function getBorderColor({ isValid, theme }: StyledProps<{ isValid: boolean; }>): string {
+    if (isValid) {
+        return theme.greys['dark-grey'];
+    }
+
+    return theme.notifications['alert-2.1'];
+}
+
+const StyledInput = styled(TextInput)`
+    flex: 1;
+    margin-bottom: 0;
+
+    input {
+        ::-ms-reveal {
+            display: none;
+        }
+
+        ::-webkit-credentials-auto-fill-button {
+            margin-right: 40px;
+        }
+
+        background-position: calc(100% - 40px) !important;
+        border-color: ${getBorderColor};
+        border-radius: var(--border-radius);
+        border-width: 1px;
+    }
+`;
+
+const StyledIconButton = styled(IconButton)<{ isValid: boolean }>`
+    background: transparent;
+    border-radius: 0 var(--border-radius) var(--border-radius) 0;
+    position: absolute;
+    right: 0;
+    top: 0;
+
+    :hover {
+        background-color: transparent;
+    }
+`;
+
+const PasswordStrengthLabel = styled.div`
+    color: ${({ theme }) => theme.greys['dark-grey']};
+    font-size: 0.75rem;
+    line-height: 1.25rem;
+`;
+
+interface PasswordCreationInputProps {
+    name?: string;
+    id?: string;
+    onChange(event: ChangeEvent<HTMLInputElement>, isValid: boolean): void;
+    validations?: ValidationCondition[];
+}
+
+export const PasswordCreationInput: VoidFunctionComponent<PasswordCreationInputProps> = ({
+    id: providedId,
+    name,
+    onChange,
+    validations,
+}) => {
+    const { t } = useTranslation('password-creation-input');
+    const [showPassword, setShowPassword] = useState(false);
+    const [password, setPassword] = useState('');
+    const isEmpty = password.length === 0;
+    const { strength: passwordStrength, label: passwordStrengthLabel } = getPasswordStrength(password, t);
+    const conditions = validations ?? getDefaultValidationConditions(t);
+    const passwordStrengthUuid = useMemo(() => uuid(), []);
+    const id = useMemo(() => providedId || uuid(), [providedId]);
+    const hintUuid = useMemo(() => uuid(), []);
+    const isValid = conditions.every((condition) => condition.isValid(password));
+
+    const handleShowPassword = (): void => {
+        setShowPassword(!showPassword);
+    };
+
+    const handleChange = (event: ChangeEvent<HTMLInputElement>): void => {
+        const newPassword = event.target.value;
+        setPassword(newPassword);
+        const newPasswordIsValid = conditions.every((condition) => condition.isValid(newPassword));
+        onChange(event, newPasswordIsValid);
+    };
+
+    return (
+        <StyledFieldContainer
```

Comment:
Le `for="..."` du `label` utilise l'`Id` du message d'erreur alors qu'il devrait utiliser celui de l'`input`